google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala

```diff
@@ -31,13 +31,27 @@ private final class GooglePubSubSource(projectId: String,
       def fetch(implicit mat: Materializer): Unit = {
         import mat.executionContext
 
-        val req = session.getToken().flatMap { token =>
-          httpApi.pull(project = projectId, subscription = subscription, accessToken = token, apiKey = apiKey)
+        val req = if (httpApi.isEmulated) {
+          val resp = httpApi
+            .pull(project = projectId, subscription = subscription, maybeAccessToken = None, apiKey = apiKey)
+          state = Fetching
```

Comment:
If you move the state change to its original spot it get even better...

Reply:
Fixed.